File: eau07.c
#include <stdbool>

bool is_white_space(char c) {
    return c == ' ' || c == '\t' || c == '\n'
}

int len(char* str) {
    int i = 0;
    while (str[i]) {
        i++;
    }

    return i;
}

int main(int argc, char* argv[]) {
    if (argc != 2) {
        return 1;
    }

    char* str = argv[1];
    int length = len(str);

    int i = 0;
    while (i < len) {
        char c = 
        while (is_white_space(str[i])) {
            i++;
        }

        i++;
    }

}

File: eau07.exs
# Majuscule a chaque mot
defmodule Exercice do
  def validate_args([arg]), do: {:ok, arg}
  def validate_args(_args), do: :error

  @spec titleize(binary()) :: binary()
  def titleize(binary) do
    do_titleize(binary, <<>>, true)
  end

  defguard is_whitespace(char) when char in [?\r, ?\n, ?\s]
  defguard is_alpha(char) when char in ?a..?z

  defp do_titleize(<<>>, acc, _upper?), do: acc

  defp do_titleize(<<char, tail::binary>>, acc, _upper?) when is_whitespace(char) do
    do_titleize(tail, <<acc::binary, char>>, true)
  end

  defp do_titleize(<<char, tail::binary>>, acc, upper?) when is_alpha(char) do
    if upper? do
      do_titleize(tail, <<acc::binary, char - 32>>, false)
    else
      do_titleize(tail, <<acc::binary, char>>, true)
    end
  end
end

System.argv()
|> Exercice.validate_args()
|> case do
  {:ok, arg} ->
    arg
    |> Exercice.titleize()
    |> IO.puts()

  :error ->
    IO.puts("usage: elixir eau07.exs <string>")
end
